Offload API to static files + Fix search bug

tuanpham96 opened the pull request:
- Get, save, and push records data to repo to query from, rather than doing API calls inline
  - Address issue #12  
  - Scheduled check and push of the records data to the repo
    - Note: to prevent too many `[bot]` commits, this step can be done & pushed in another repo  
  - Sanitize parsed README html
- Fix search bug with `miniscope` turns to nothing issue #8 
  - I think `*` is not exactly equivalent to regex `.*` but instead `.+`
  - Use fuzzy match of edit distance 1 to mediate this

tuanpham96 committed:
add script to get data

tuanpham96 committed:
modify home to read from data/records.json

tuanpham96 committed:
Create jekyll.yml

This was autocreated by Github

tuanpham96 committed:
include explicit asset paths in config

tuanpham96 committed:
fix url paths

tuanpham96 committed:
ci for downloading records

Commit:
[bot] update records

tuanpham96 committed:
rename and clean records->data workflow

- dont run this again when the bot pushes

Commit:
[bot] update records

tuanpham96 committed:
fix the weird partial search bug with fuzzy match

rly commented:
This is great! Thank you for the PR @tuanpham96 . It will take me a little time to review this and understand what is going on. I will either get to it by the end of the week or ping you during the NWB developer days to walk through it together.

tuanpham96 committed:
refine search with conditions

note: this is because fuzzy match with diff 1 is not enough

tuanpham96 committed:
specify header font size of readme-preview

tuanpham96 committed:
tentative: add readme in details + add pypi badges to show most updated version

Commit:
[bot] update records

Commit:
[bot] update records

tuanpham96 committed:
pass headers with requests using token

rly commented on .github/workflows/data.yml:
```diff
@@ -0,0 +1,51 @@
+on:
+  push:
+    branches: ["main"]
+  schedule:
+    # Run at 8:00 on working days
+    - cron: 0 8 * * 0-5
```
```suggestion
  schedule:
    # Run at 8:00 on working days
    - cron: 0 8 * * 0-5
  workflow_dispatch:
```

It would be nice to run this on command too.

rly commented on .github/workflows/data.yml:
```diff
@@ -0,0 +1,51 @@
+      CI_PUSH_REMOTE: git@github.com:$GITHUB_REPOSITORY.git
+      CI_PUSH_BRANCH: 'main'
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
+        persist-credentials: true
+    - uses: actions/setup-python@v4
+      with:
+          python-version: '3.10'
+          cache: 'pip'
```
```suggestion
    - uses: actions/checkout@v4
      with:
        ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
        persist-credentials: true
    - uses: actions/setup-python@v5
      with:
          python-version: '3.12'
          cache: 'pip'
```
Updating to latest versions.

rly commented on .github/workflows/data.yml:
```diff
@@ -0,0 +1,51 @@
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
```
To be clear, for this to work, I will have to add this secret, right?

tuanpham96 replied:
yes. this is using a project-level deploy key. An alternative is to separate the scheduled data CI into another repo and this repo would just get the data from there

rly replied:
Great, thanks! I'll set up the secret.

rly replied:
I think I can use 
```suggestion
        token: ${{ secrets.NWB_EXTENSIONS_BOT_GITHUB_TOKEN }}
```

rly commented on js/search.js:
```diff
@@ -75,14 +75,15 @@ excluded_in_search: true
         return resultHtml;
       }
-
+https://img.shields.io/pypi/v/ndx-odor-metadata?label=pypi%20package&style=flat-square&color=blue)](https://pypi.org/pypi/ndx-odor-metadata)
```
```suggestion

```
I think this snuck in

tuanpham96 replied:
oops, sorry I may have copied and pasted it things in the wrong place.

rly commented:
Thank you so much @tuanpham96 and sorry it has taken me forever to review this. I tested this locally and it looks great. I made a few suggested edits. I could also make these changes and update the `records.json` in this PR if you check "Allow edits from maintainers".

tuanpham96 committed:
Apply suggestions from code review

- add `workflow_dispatch` to run CI command 
- update versions in CI
- remove irrelevant link

Co-authored-by: Ryan Ly <rly@lbl.gov>

Commit:
[bot] update records

tuanpham96 commented:
@rly thanks for reviewing! I just made your suggested changes, and also checked the "Allow edits" button.

rly committed:
Update .github/workflows/data.yml

rly commented:
Looks good. Thank you! Let's try it out and debug any issues from there.
